diff -Nuar a/build/cmake/CMakeLists.txt b/build/cmake/CMakeLists.txt
--- a/build/cmake/CMakeLists.txt	2020-01-23 04:02:19.297625325 +0300
+++ b/build/cmake/CMakeLists.txt	2020-01-23 04:02:57.444622827 +0300
@@ -109,6 +109,8 @@
   target_compile_options (glew_s PRIVATE -GS-)
   # remove stdlib dependency
   target_link_libraries (glew LINK_PRIVATE -nodefaultlib -noentry)
+  target_link_libraries (glew LINK_PRIVATE libvcruntime.lib)
+  target_link_libraries (glew LINK_PRIVATE msvcrt.lib )
   string(REGEX REPLACE "/RTC(su|[1su])" "" CMAKE_C_FLAGS_DEBUG ${CMAKE_C_FLAGS_DEBUG})
 elseif (WIN32 AND ((CMAKE_C_COMPILER_ID MATCHES "GNU") OR (CMAKE_C_COMPILER_ID MATCHES "Clang")))
   # remove stdlib dependency on windows with GCC and Clang (for similar reasons
