--- a/src/3rdparty/chromium/v8/src/torque/implementation-visitor.cc
+++ b/src/3rdparty/chromium/v8/src/torque/implementation-visitor.cc
@@ -54,6 +54,9 @@
     std::ostream& source = GlobalContext::GeneratedPerFile(file).csa_ccfile;
     std::ostream& header = GlobalContext::GeneratedPerFile(file).csa_headerfile;
 
+    source << "#include \"src/objects/cell.h\"\n";
+    source << "\n";
+
     for (const std::string& include_path : GlobalContext::CppIncludes()) {
       source << "#include " << StringLiteralQuote(include_path) << "\n";
     }
@@ -3015,9 +3018,12 @@
   std::string base_name = "internal-class-definitions-tq";
   {
     IncludeGuardScope header_guard(header, base_name + ".h");
+    header << "#include \"src/objects/cell.h\"\n";
     header << "#include \"src/objects/objects.h\"\n";
     header << "#include \"src/objects/struct.h\"\n";
     header << "#include \"src/objects/js-objects.h\"\n";
+    header << "#include \"src/objects/js-promise.h\"\n";
+    header << "#include \"src/objects/js-generator.h\"\n";
     header << "#include \"src/utils/utils.h\"\n";
     header << "#include \"torque-generated/class-definitions-tq.h\"\n";
     IncludeObjectMacrosScope header_macros(header);
@@ -3717,6 +3723,45 @@
     h_contents << "#include \"torque-generated/csa-types-tq.h\"\n";
     h_contents
         << "#include \"torque-generated/internal-class-definitions-tq.h\"\n";
+
+    cc_contents << "#include \"src/objects/free-space.h\"\n";
+    cc_contents << "#include \"src/objects/js-regexp-string-iterator.h\"\n";
+    cc_contents << "#include \"src/objects/ordered-hash-table.h\"\n";
+    cc_contents << "#include \"src/objects/property-descriptor-object.h\"\n";
+    cc_contents << "#include \"src/objects/synthetic-module.h\"\n";
+    cc_contents << "#include \"src/objects/template-objects.h\"\n";
+    cc_contents << "#include \"src/objects/templates.h\"\n";
+    cc_contents << "#include \"src/objects/stack-frame-info.h\"\n";
+    cc_contents << "#include \"src/objects/source-text-module.h\"\n";
+    cc_contents << "#include \"src/objects/property-cell.h\"\n";
+    cc_contents << "#include \"src/objects/literal-objects.h\"\n";
+    cc_contents << "#include \"src/objects/js-weak-refs.h\"\n";
+    cc_contents << "#include \"src/objects/js-generator.h\"\n";
+    cc_contents << "#include \"src/objects/feedback-vector.h\"\n";
+    cc_contents << "#include \"src/objects/embedder-data-array.h\"\n";
+    cc_contents << "#include \"src/objects/cell.h\"\n";
+    cc_contents << "#include \"src/objects/api-callbacks.h\"\n";
+    cc_contents << "#include \"src/objects/descriptor-array.h\"\n";
+    cc_contents << "#include \"src/objects/feedback-cell.h\"\n";
+    cc_contents << "#include \"src/objects/js-promise.h\"\n";
+    cc_contents << "#include \"src/objects/regexp-match-info.h\"\n";
+    cc_contents << "#include \"src/objects/prototype-info.h\"\n";
+    cc_contents << "#include \"src/objects/scope-info.h\"\n";
+    cc_contents << "#include \"src/wasm/wasm-objects.h\"\n";
+    {
+      IfDefScope intl_scope(cc_contents, "V8_INTL_SUPPORT");
+      cc_contents << "#include \"src/objects/js-break-iterator.h\"\n";
+      cc_contents << "#include \"src/objects/js-collator.h\"\n";
+      cc_contents << "#include \"src/objects/js-date-time-format.h\"\n";
+      cc_contents << "#include \"src/objects/js-display-names.h\"\n";
+      cc_contents << "#include \"src/objects/js-list-format.h\"\n";
+      cc_contents << "#include \"src/objects/js-locale.h\"\n";
+      cc_contents << "#include \"src/objects/js-number-format.h\"\n";
+      cc_contents << "#include \"src/objects/js-plural-rules.h\"\n";
+      cc_contents << "#include \"src/objects/js-relative-time-format.h\"\n";
+      cc_contents << "#include \"src/objects/js-segment-iterator.h\"\n";
+      cc_contents << "#include \"src/objects/js-segmenter.h\"\n";
+    }
     cc_contents << "#include \"torque-generated/" << file_name << ".h\"\n";
 
     for (SourceId file : SourceFileMap::AllSources()) {
