diff --git a/src/common/dwarf/functioninfo.cc b/src/common/dwarf/functioninfo.cc
index 358a6eef..1c974b43 100644
--- a/src/common/dwarf/functioninfo.cc
+++ b/src/common/dwarf/functioninfo.cc
@@ -69,13 +69,13 @@ void CULineInfoHandler::DefineDir(const string& name, uint32_t dir_num) {
 }
 
 void CULineInfoHandler::DefineFile(const string& name,
-                                   int32 file_num, uint32_t dir_num,
+                                   int32_t file_num, uint32_t dir_num,
                                    uint64_t mod_time, uint64_t length) {
   assert(dir_num >= 0);
   assert(dir_num < dirs_->size());
 
   // These should never come out of order, actually.
-  if (file_num == (int32)files_->size() || file_num == -1) {
+  if (file_num == static_cast<int32_t>(files_->size()) || file_num == -1) {
     string dir = dirs_->at(dir_num);
 
     SourceFileInfo s;
diff --git a/src/common/dwarf/functioninfo.h b/src/common/dwarf/functioninfo.h
index 5c733c6d..0cf3aca4 100644
--- a/src/common/dwarf/functioninfo.h
+++ b/src/common/dwarf/functioninfo.h
@@ -69,8 +69,8 @@ struct SourceFileInfo {
   uint64_t lowpc;
 };
 
-typedef std::map<uint64, FunctionInfo*> FunctionMap;
-typedef std::map<uint64, std::pair<string, uint32> > LineMap;
+typedef std::map<uint64_t, FunctionInfo*> FunctionMap;
+typedef std::map<uint64_t, std::pair<string, uint32_t> > LineMap;
 
 // This class is a basic line info handler that fills in the dirs,
 // file, and linemap passed into it with the data produced from the
@@ -90,7 +90,7 @@ class CULineInfoHandler: public LineInfoHandler {
 
   // Called when we define a filename.  We just place
   // concat(dirs_[DIR_NUM], NAME) into files_ at position FILE_NUM.
-  virtual void DefineFile(const string& name, int32 file_num,
+  virtual void DefineFile(const string& name, int32_t file_num,
                           uint32_t dir_num, uint64_t mod_time, uint64_t length);
 
 
