diff --git a/util/BUILD.gn b/util/BUILD.gn
index 21f70048..c4724f8d 100644
--- a/crashpad/util/BUILD.gn
+++ b/crashpad/util/BUILD.gn
@@ -379,8 +379,8 @@ static_library("util") {
         deps += [ "//third_party/boringssl" ]
       } else {
         libs = [
-          "crypto",
           "ssl",
+          "crypto",
         ]
       }
     }
@@ -637,8 +637,8 @@ if (!crashpad_is_android && !crashpad_is_ios) {
         deps += [ "//third_party/boringssl" ]
       } else {
         libs = [
-          "crypto",
           "ssl",
+          "crypto",
         ]
       }
     }
