diff --git a/build/BUILD.gn b/build/BUILD.gn
index d8dc6cd..4f55b5f 100644
--- a/build/BUILD.gn
+++ b/build/BUILD.gn
@@ -39,10 +39,23 @@ if (mini_chromium_is_mac) {
     # win_sdk\bin\SetEnv.cmd inside this path will be used to configure the
     # Windows toolchain.
     win_toolchain_path = "<autodetect>"
+
+    # linktime optimization (enables compiler flag /GL and linker flag /LTCG)
+    linktime_optimization = true
+
+    # Link CRT dynamically or statically
+    dynamic_crt = false
   }
 }
 
 config("debug") {
+  if (mini_chromium_is_win) {
+    if (dynamic_crt) {
+      cflags = [ "/MDd" ]
+    } else {
+      cflags = [ "/MTd" ]
+    }
+  }
 }
 
 config("release") {
@@ -64,7 +77,6 @@ config("release") {
     }
   } else if (mini_chromium_is_win) {
     cflags = [
-      "/GL",  # LTCG.
       "/O2",
       "/Ob2",  # Both explicit and auto inlining.
       "/Oy-",  # Disable omitting frame pointers, must be after /O2.
@@ -73,10 +85,20 @@ config("release") {
     ]
     ldflags = [
       "/OPT:ICF",
-      "/OPT:REF",
-      "/LTCG",
+      "/OPT:REF"
     ]
-    arflags = [ "/LTCG" ]
+
+    if (linktime_optimization) {
+      cflags += [ "/GL" ] # LTCG
+      ldflags += [ "/LTCG" ]
+      arflags = [ "/LTCG" ]
+    }
+
+    if (dynamic_crt) {
+      cflags += [ "/MD" ]
+    } else {
+      cflags += [ "/MT" ]
+    }
   }
 }
 
