--- a/libbase/logging.cpp	2021-07-14 10:01:54.373014821 +0200
+++ b/libbase/logging.cpp	2021-07-14 10:03:04.200476140 +0200
@@ -314,6 +314,7 @@
   abort();
 }
 
+#ifdef __ANDROID__
 static void LogdLogChunk(LogId id, LogSeverity severity, const char* tag, const char* message) {
   int32_t lg_id = LogIdTolog_id_t(id);
   int32_t priority = LogSeverityToPriority(severity);
@@ -337,6 +338,7 @@
 
   SplitByLogdChunks(id, severity, tag, file, line, message, LogdLogChunk);
 }
+#endif
 
 void InitLogging(char* argv[], LogFunction&& logger, AbortFunction&& aborter) {
   SetLogger(std::forward<LogFunction>(logger));