diff --git a/ports/qt5-base/portfile.cmake b/ports/qt5-base/portfile.cmake
index 443d462e1..cc6403d19 100644
--- a/ports/qt5-base/portfile.cmake
+++ b/ports/qt5-base/portfile.cmake
@@ -68,6 +68,7 @@ set(CORE_OPTIONS
     # ENV ANGLE_DIR to external angle source dir. (Will always be compiled with Qt)
     #-optimized-tools
     #-force-debug-info
+    -no-use-gold-linker
     -verbose
 )
 
@@ -82,6 +83,7 @@ list(APPEND CORE_OPTIONS
     -system-sqlite
     -system-harfbuzz
     -icu
+    -no-glib
     -no-angle)      # Qt does not need to build angle. VCPKG will build angle!
 
 if(QT_OPENSSL_LINK)
